src/ProjectTemplates/BlazorWasm.ProjectTemplates/content/BlazorWasm-CSharp/Client/Services/IdentityAuthenticationStateProvider.cs

```diff
@@ -0,0 +1,59 @@
+using BlazorWasm_CSharp.Shared.Models;
+using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Components.Authorization;
+using System.Net.Http;
+using System.Security.Claims;
+using System.Threading.Tasks;
+
+namespace BlazorWasm_CSharp.Client
+{
+    public class IdentityAuthenticationStateProvider : AuthenticationStateProvider
+    {
+        private ClaimsPrincipal _currentUser;
+
+        public IdentityAuthenticationStateProvider(HttpClient client)
+        {
+            Client = client;
+        }
+
+        public HttpClient Client { get; }
```

Comment:
Does this need to be public? Could it be a private field?

Reply:
Updated.